app/src/org/commcare/views/widgets/RecordingFragment.java

```diff
@@ -0,0 +1,330 @@
+        saveRecording.setVisibility(View.INVISIBLE);
+        recordAgain.setVisibility(View.INVISIBLE);
+        recordingDuration.setVisibility(View.INVISIBLE);
+        GoogleAnalyticsUtils.reportRecordingRecycled();
+    }
+
+    private void startRecording(){
+        disableScreenRotation();
+        setCancelable(false);
+
+        setupRecorder();
+        recorder.start();
+
+        toggleRecording.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                GoogleAnalyticsUtils.reportRecordingStopped();
+                stopRecording();
+            }
+        });
+        toggleRecording.setBackgroundResource(R.drawable.record_in_progress);
+        instruction.setText(Localization.get("during.recording"));
+
+        recordingProgress.setVisibility(View.VISIBLE);
+        recordingDuration.setVisibility(View.VISIBLE);
+
+        recordingDuration.setBase(SystemClock.elapsedRealtime());
+        recordingDuration.start();
+    }
+
+    private void setupRecorder() {
+        if(recorder == null){
+            recorder = new MediaRecorder();
+        }
+
+        recorder.setAudioSource(MediaRecorder.AudioSource.MIC);
+        recorder.setOutputFormat(MediaRecorder.OutputFormat.MPEG_4);
+        recorder.setOutputFile(fileName);
+        recorder.setAudioEncoder(MediaRecorder.AudioEncoder.AMR_NB);
+        try{
+            recorder.prepare();
+        }catch(IOException e){
+            Log.d("Recorder", "Failed to prepare media recorder");
+        }
+    }
+
+    private void stopRecording(){
+
+        recordingDuration.stop();
+        recordAgain.setVisibility(View.VISIBLE);
+        recordingProgress.setVisibility(View.INVISIBLE);
+        recorder.stop();
+        toggleRecording.setBackgroundResource(R.drawable.play);
+        toggleRecording.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                playAudio();
+                GoogleAnalyticsUtils.reportAudioPlayed();
+            }
+        });
+        saveRecording.setEnabled(true);
+        saveRecording.setVisibility(View.VISIBLE);
+        instruction.setText(Localization.get("after.recording"));
+    }
+
+    public void saveRecording(){
+        if(listener != null){
+            listener.onRecordingCompletion();
+        }
+        dismiss();
+    }
+
+    public interface RecordingCompletionListener {
+        void onRecordingCompletion();
+    }
+
+    public void setListener(RecordingCompletionListener listener){
+        this.listener = listener;
+    }
+
+    @Override
+    public void onDismiss(DialogInterface dialog){
+        super.onDismiss(dialog);
+        enableScreenRotation();
+        if(recorder != null){
+            recorder.release();
+            this.recorder = null;
+        }
+    }
+
+    public String getFileName(){
+        return fileName;
+    }
+
+    private void disableScreenRotation() {

```

Comment:
This can be made into a `static` method that takes in the activity as an argument. That way it can be reused in the calendar code as well